src/get-started/flutter-for/declarative.md

```diff
@@ -68,3 +68,27 @@ RenderObjects persist between frames and Flutter’s lightweight Widgets
 tell the framework to mutate the RenderObjects between states.
 The Flutter framework handles the rest.
 
+## Declarative extension style
+
+There is a new declarative pattern that has emerged as a result of Dart 
+adding the `extension` language feature. While this pattern is not currently 
+in use in the Flutter framework, it is actively in use within the community.
+
+This declarative style does appear at first glance to be similar to the imperative
+style shown above, but the underlying operation is composing widgets through
+their constructor just like in the declarative style shown above.
```

Comment:
**issues:**
- Write in active voice per [GSG](https://developers.google.com/style/voice).
- Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading).

```suggestion
The Flutter community surfaced a new declarative pattern as a result of Dart 
adding the `extension` language feature. The Flutter framework doesn't use
this pattern at this time. The community does use this pattern.

This declarative style does appear to resemble the imperative style that
other frameworks use. However, the underlying operation composes
widgets through their constructor just like in the [declarative style](#why-a-declarative-ui).
```

Reply:
As far as general Dart style is concerned, sure, there's nothing wrong with using extensions. As to whether that's good style in Flutter code and/or worth documenting it on Flutter's own docs... that's outside of my area of expertise.

I do find this section confusing. It's responding to an argument that isn't stated on the doc itself. Anyone who comes to this page who *hasn't* already heard someone claim that extension methods aren't declarative will have little context as to why the doc is justifying them and be confused as to what it's trying to get at.

Also, this section emphasizes extension methods, but it's not the fact that these are *extension* methods that makes them declarative. You can have imperative extension methods and declarative instance methods:

```dart
extension ListExt<T> on List<T> {
  void imperative(T t) {
    add(t);
    add(t);
  }
}

class MyList<T> extends ListBase<T> {
  List<T> declarative(T t) => [...this, t, t];
}
```

What matters is whether the method is *mutating* versus creating a new object.